include/xrpl/protocol/SField.h

```diff
@@ -520,7 +551,7 @@ extern SF_UINT256 const sfHookNamespace;
 extern SF_UINT256 const sfHookSetTxnID;
 
 // currency amount (common)
-extern SF_AMOUNT const sfAmount;
+extern SF_EITHER_AMOUNT const sfAmount;
```

Comment:
why wouldn't we need to change the other Amount types like sfBalance to SF_EITHER_AMOUNT?

Reply:
`SF_EITHER_AMOUNT` is used to indicate that this field can be either `STAmount` or `STMPTAmount`. `SF_AMOUNT` indicates that the field can be only `STAmount`. Only `sfAmount` supports MPT in the first release. What this actually does is this: when you access `sfAmount` as `tx[sfAmount]` then it returns `STEitherAmount` and you have explicitly get the amount with `get<>()`. If you access `sfBalance` as `tx[sfBalance]` then it returns `STAmount`.

Reply:
Yes I understand that. But don't we already have something like

Reply:
Yes I understand that, but don't we already have something like this inside the TxFormat.cpp?
`{sfAmount, soeREQUIRED, soeMPTSupported},`

Or i guess this one is compile-type check where as the SOElement is runtime.